src/Microsoft.VisualStudio.Threading.Analyzers/AbstractVSTHRD012SpecifyJtfWhereAllowed.cs

```diff
@@ -81,8 +81,10 @@ private static void AnalyzeCall(OperationAnalysisContext context, Location locat
             // The method being invoked doesn't take any JTC/JTF parameters.
             // Look for an overload that does.
             bool preferableAlternativesExist = otherOverloads
-                .Where(m => !m.IsObsolete())
-                .Any(m => m.Parameters.Skip(m.IsExtensionMethod ? 1 : 0).Any(IsImportantJtfParameter));
+                .Any(m =>
+                    !m.IsObsolete() &&
+                    m.Parameters.Skip(m.IsExtensionMethod ? 1 : 0).Any(IsImportantJtfParameter) &&
+                    context.ContainingSymbol.FindContainingNamedOrAssemblySymbol() is ISymbol containingSymbol && context.Compilation.IsSymbolAccessibleWithin(m, containingSymbol));
```

Comment:
consider splitting each check onto it's own line.

Reply:
I generally like what you're proposing. This pair in particular share a line because it's really the second check that matters, but the first check is a special check that must be met before the second check can be allowed. All the 'lines' here could be in any order, but are sorted in order of runtime execution cost. So by putting these two checks on the same line, I make it easier to see that they are connected and in a particularly required order.